### DISCUSSION TITLE
Link objects appearing as broken links on first load

### DISCUSSION THREAD

**Discussion by Delca** [mozilla/hubs#4729] [Hubs Cloud Questions]
While trying to setup a scene in an AWS Hubs Cloud environment I encountered an issue where my link objects appear as broken links when the scene is first loaded, only to get resolved normally if I use the `Refresh` button on them later on.

The scene itself has around 30 links objects, one being a video stream and the rest linking to HTML pages. Looking at the server code and logs it seems like as soon as someone connects to the room all 30-something requests for the media content are fired to the server, to then eventually get processed through `youtube-dl` until the client gets back a 500 Server Error response and decided to display the object as a broken link.
This processing goes on for at least a minute or two after this single user joined the room, and it looks like the `youtube-dl` service can only handle (or is only served) one request at a time.

By going to each link object in the room and manually updating them with the `Refresh` button one by one I can see that every link then gets processed correctly, which to me proves that the issues does not lie within the links themselves but rather in the number of requests that is made simultaneously.
Since this scene is still only a prototype that contains less than half of the content I am trying to integrate in it, I would like to know if what I am trying to achieve is a realistic goal to target with the Hubs environment, and what solutions I could try to fix the issues I am (or will) encounter with this feature

**Comment by pattersonbl2** [comment]
could you tell me more about where the links are generated from ?   Are these a bunch of video links from Youtube ?

**Reply by Delca** [reply]
Sure. The HTML links themselves points to various product gallery pages; they do not contain video content and the only reference to YouTube in them is a link to a channel in the page's footer. Here are two of the links we are using in the scene:

https://meeco.mistore.jp/contents/paul_joe_beaute/index.html
https://meeco.mistore.jp/contents/three/index.html

The video has been uploaded through Spoke and is served through our Hubs instance.

**Comment by Charlesc22** [comment]
We've seen this behavior as well for a long time. And always suspected it was a sort of queuing issue where if the queue gets too backed up something times out/gives up and just stops working and decides that all the links are broken.

**Reply by pattersonbl2** [reply]
From what you're saying it looks like youtube-dl is the central problem.  I do know that youtube-dl will process all videos and convert them to a file format that is readable by hubs across the board. This may be tricky to fix because youtube-dl is open source code that isn't actively managed by us.  Can you send me the logs where you are seeing the queue hang so I can do a deeper dive with my team?

**Reply by Delca** [reply]
@Pattersonbl2 I sent you an email with a recording of the app server logs during a simple connection to a room. While the 35 requests to the media API eventually get a 500 response after 15 seconds, we can see errors related to youtube-dl signaling unsupported URLs up to 100 seconds after the initial requests were observed, which suggest to me that they are getting queued up somewhere.

**Reply by pattersonbl2** [reply]
https://github.com/mozilla/reticulum/blob/master/lib/ret/media_resolver.ex#L173-L215    You could take a deep dive into the code as well to see if something here could help.

**Reply by sea922** [reply]
@Delca  I had the same problem, did you find the solution?
